var/spack/repos/builtin/packages/unifyfs/package.py

```diff
@@ -16,55 +16,50 @@ class Unifyfs(AutotoolsPackage):
 
     homepage = "https://github.com/LLNL/UnifyFS"
     git      = "https://github.com/LLNL/UnifyFS.git"
-    url      = "https://github.com/LLNL/UnifyFS/releases/download/v0.9.0/unifyfs-0.9.0.tar.gz"
+    url      = "https://github.com/LLNL/UnifyFS/releases/download/v0.9.2/unifyfs-0.9.2.tar.gz"
     maintainers = ['CamStan']
 
     version('develop', branch='dev')
+    version('0.9.2', sha256='7046625dc0677535f5d960187cb2e2d58a6f8cfb4dc6a3604f825257eb0891aa')
     version('0.9.1', sha256='2498a859cfa4961356fdf5c4c17e3afc3de7e034ad013b8c7145a622ef6199a0')
-    version('0.9.0', sha256='e6c73e22ef1c23f3141646aa17058b69c1c4e526886771f8fe982da924265b0f')
+    version('0.9.0', sha256='e6c73e22ef1c23f3141646aa17058b69c1c4e526886771f8fe982da924265b0f', deprecated=True)
 
     variant('auto-mount', default='True', description='Enable automatic mount/unmount in MPI_Init/Finalize')
     variant('hdf5', default='False', description='Build with parallel HDF5 (install with `^hdf5~mpi` for serial)')
     variant('fortran', default='False', description='Build with gfortran support')
-    variant('mdhim', default='False', description='Enable MDHIM build options')
     variant('pmi', default='False', description='Enable PMI2 build options')
     variant('pmix', default='False', description='Enable PMIx build options')
     variant('spath', default='True', description='Use spath library to normalize relative paths')
 
-    depends_on('autoconf',  type='build')
-    depends_on('automake',  type='build')
-    depends_on('libtool',   type='build')
-    depends_on('m4',        type='build')
-    depends_on('pkgconfig', type='build')
+    depends_on('autoconf',        type='build')
+    depends_on('automake',        type='build')
+    depends_on('automake@1.15:',  type='build', when='@0.9.2:')
+    depends_on('libtool',         type='build')
+    depends_on('m4',              type='build')
+    depends_on('pkgconfig',       type='build')
 
     # Required dependencies
-    depends_on('gotcha@1.0.3:', when='@0.9.1:')
-    depends_on('mochi-margo@0.4.3')
-    depends_on('mercury@1.0.1+bmi+sm')
+    depends_on('gotcha@1.0.3:')
+    depends_on('mercury@1.0.1+bmi', when='@:0.9.1')
+    depends_on('mochi-margo@0.4.3', when='@:0.9.1')
+    depends_on('mochi-margo',       when='@0.9.2:')
     depends_on('mpi')
     depends_on('openssl')
 
     # Optional dependencies
     depends_on('hdf5', when='+hdf5')
-    depends_on('leveldb', when='@0.9.1:+mdhim')
-    depends_on('spath', when='@0.9.1:+spath')
+    depends_on('libfabric fabrics=rxm,sockets,tcp', when="^mercury@2:+ofi")
+    depends_on('libfabric@1.9.1', when="^mochi-margo@0.4.3 ^mercury+ofi")
+    depends_on('mercury@1.0.1', when='^mochi-margo@0.4.3')
```

Comment:
This looks duplicated given:

```
depends_on('mercury@1.0.1+bmi', when='@:0.9.1')
depends_on('mochi-margo@0.4.3', when='@:0.9.1')
```

earlier

Reply:
Not exactly duplicate, but probably unnecessary.
This
> ```
> depends_on('mercury@1.0.1+bmi', when='@:0.9.1')
> depends_on('mochi-margo@0.4.3', when='@:0.9.1')
> ```
is what we need for version `0.9.1` as that version breaks with later versions of margo/mercury.

These:
```
depends_on('libfabric@1.9.1', when="^mochi-margo@0.4.3 ^mercury+ofi")
depends_on('mercury@1.0.1', when='^mochi-margo@0.4.3')
```
were intended to say that we still support mochi-margo version `0.4.3` with our newest release if a user wanted/needed to fallback to it, and this would get them the proper versions of mercury and libfabric should they do so.
E.g., `spack install unifyfs@0.9.2 ^mochi-margo@0.4.3`

However, that is definitely an unlikely scenario and we'd prefer users to use the latest versions of mochi-margo/mercury anyways. So I'm thinking I should still pull these out, yeah?

Reply:
Great points. I was definitely wrong that the extra constraint was redundant. FWIW I think in the Spack package it is "OK" to simplify the constraints to build a subset of all the possible things. Feel free to leave this in though (especially if you think users will want this arrangement).

Reply:
I doubt users really need this arrangement, to be honest. It was more in case we needed them to fallback to a stable scenario. If that's the case though, they'll already be in contact with us and we can always tell them to try:
```
spack install unifyfs ^mochi-margo@0.4.3 ^mercury@1.0.1 ^libfabric@1.9.1
```
or even just fallback to v0.9.1.

However, I think it's highly unlikely this will come up. I think leaning towards simplifying the Spack package is the best option here and I'll go ahead and pull these out.